### PR TITLE
defaults: add `controlcenter` and other options

### PR DESCRIPTION
**Pull request by LucasMendesl**
Adding support to control center API

### CODE REVIEW

File: modules/system/defaults/controlcenter.nix [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        default = null; 
+        description = ''
+            Apple menu > System Preferences > Control Center > Battery
+
+            Show a battery percentage in menu bar. Default is null.
+        '';
+    };
+
+    system.defaults.controlcenter.Sound = mkOption {
+        type = types.nullOr (types.enum [ 18 24 ]);
```
Comment (Enzime) [inline modules/system/defaults/controlcenter.nix:19]:
```suggestion
        type = types.nullOr types.bool;
        apply = v: if v == null then null else if v == true then 18 else 24;
```

**Review by Enzime — CHANGES_REQUESTED** [review]

File: modules/system/defaults-write.nix [hunk 2]
```diff
@@ -17,6 +17,7 @@ let
   SoftwareUpdate = defaultsToList "/Library/Preferences/com.apple.SoftwareUpdate" cfg.SoftwareUpdate;
 
   # userDefaults
+  controlcenter = defaultsToList "${builtins.getEnv "HOME"}/Library/Preferences/ByHost/com.apple.controlcenter" cfg.controlcenter;
```
[the remark this answers is not in the document]
Comment (LucasMendesl) [reply modules/system/defaults-write.nix]: using the complete path of this domain is one approach to change this setting, another approach is using the flag `-currentHost` like:

```sh
defaults -currentHost write com.apple.controlcenter BatteryShowPercentage -bool true
```

File: modules/system/defaults/controlcenter.nix [hunk 3]
```diff
@@ -0,0 +1,96 @@
+{ config, lib, ... }:
+
+with lib;
```
[the remark this answers is not in the document]
Comment (LucasMendesl) [reply modules/system/defaults/controlcenter.nix]: Changed

File: modules/system/defaults/controlcenter.nix [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+{
+  options = {
+
+    system.defaults.controlcenter.BatteryShowPercentage = mkOption {
```
[the remark this answers is not in the document]
Comment (LucasMendesl) [reply modules/system/defaults/controlcenter.nix]: Changed

File: modules/system/defaults/controlcenter.nix (hunk 1 above)
Comment (LucasMendesl) [reply modules/system/defaults/controlcenter.nix]: Changed

**Comment by Enzime** [comment]
You'll need to add your new module file to `modules/module-list.nix`

**Comment by Enzime** [comment]
You'll also need to add `controlcenter` to here: https://github.com/LnL7/nix-darwin/pull/1163/files#diff-b44750cda15d1291f935ca5053d5aa14c6b1492c38c8bd9c078a0b2d752eaad6L72

And you'll need to update `tests/fixtures/system-defaults-write/activate-user.txt`

**Comment by LucasMendesl** [comment]
@Enzime, fixed issues and including #1107

**Comment by LucasMendesl** [comment]
> You'll need to add your new module file to `modules/module-list.nix`

Done

**Comment by Enzime** [comment]
Can you squash it into 2 commits? One for the `controlcenter` changes and another one for the non-`controlcenter` changes

**Comment by LucasMendesl** [comment]
> Can you squash it into 2 commits? One for the `controlcenter` changes and another one for the non-`controlcenter` changes

@Enzime, Done 😄

File: modules/system/defaults-write.nix [hunk 5]
```diff
@@ -71,6 +72,7 @@ in
 
     system.activationScripts.userDefaults.text = mkIfAttrs
       [
+        controlcenter
```
[the remark this answers is not in the document]
Comment (LucasMendesl) [reply modules/system/defaults-write.nix]: Done

**Review by Enzime — APPROVED** [review]
LGTM

Thanks for the contribution

**Pull request merged**